api/v1alpha1/mcpserver_types.go

```diff
@@ -210,6 +223,87 @@ type MCPServerDeployment struct {
 	SecretRefs []corev1.ObjectReference `json:"secretRefs,omitempty"`
 }
 
+// MCPServerAuthentication defines the authentication configuration for the MCP server.
+type MCPServerAuthentication struct {
+	// JWT defines the JWT authentication configuration.
+	JWT *MCPServerJWTAuthentication `json:"jwt,omitempty"`
+}
+
+// MCPServerJWTAuthentication defines the JWT authentication configuration for the MCP server.
+type MCPServerJWTAuthentication struct {
+	// Issuer is the JWT issuer URL.
+	Issuer string `json:"issuer,omitempty"`
+
+	// Audiences is a list of audiences that the JWT must match.
+	Audiences []string `json:"audiences,omitempty"`
+
+	// JWKS references a secret containing the JSON Web Key Set.
+	// The secret must contain a key with the JWKS content.
+	JWKS *corev1.SecretKeySelector `json:"jwks,omitempty"`
+}
+
+// MCPServerAuthorization defines the authorization configuration for the MCP server.
+type MCPServerAuthorization struct {
+	// Server defines the configuration for the MCP authorization server that protects the MCP server.
+	// Setting this field will configure agentgateway to use the authorization server
+	// to protect the MCP server and its resources as well as adapt traffic to the MCP client to comply with the
+	// MCP authorization spec before forwarding traffic to the MCP client.
+	// +optional
+	Server *MCPAuthorizationServer `json:"server,omitempty"`
+
+	// CELAuthorization defines the CEL-based authorization configuration for the MCP server.
+	CEL *MCPServerCELAuthorization `json:"cel,omitempty"`
```

Comment:
`CEL` is an implementation detail here, not a feature. Can we name this something about what it's doing, rather than how it's implemented?